pkg/tsdb/cloudwatch/log_query.go

```diff
@@ -73,9 +73,24 @@ func logsResultsToDataframes(response *cloudwatchlogs.GetQueryResultsOutput) (*d
 				timeField[i] = &parsedTime
 			} else if numericField, ok := fieldValues[*resultField.Field].([]*float64); ok {
 				parsedFloat, err := strconv.ParseFloat(*resultField.Value, 64)
+
+				// This can happen if a field has a mix of numeric and non-numeric values.
+				// In that case, we change the field from a numeric field to a string field.
 				if err != nil {
-					return nil, err
+					fieldValuesAsStrings := make([]*string, rowCount)
+					for j := 0; j <= i; j++ {
+						for _, rf := range nonEmptyRows[j] {
+							if *rf.Field == *resultField.Field {
+								fieldValuesAsStrings[j] = rf.Value
+							}
+						}
+					}
+
+					fieldValues[*resultField.Field] = fieldValuesAsStrings
+
+					continue
```

Comment:
```suggestion
					
					// This can happen if a field has a mix of numeric and non-numeric values.
					// In that case, we change the field from a numeric field to a string field.
					fieldValues[*resultField.Field] = changeToStringField(rowCount, nonEmptyRows[:i+1], *resultField.Field)
					continue
```

This is great!
What do you think about extracting this to another function? The title of the function can be very self-explanatory like your comment above. 
This will prevent this parent function from getting even bigger... 


Then the following function could go a little below.
```
func changeToStringField(lengthOfValues int, rows [][]*cloudwatchlogs.ResultField, logEventField string) []*string {
	fieldValuesAsStrings := make([]*string, lengthOfValues)
	for i, resultFields := range rows {
		for _, field := range resultFields {
			if *field.Field == logEventField {
				fieldValuesAsStrings[i] = field.Value
			}
		}
	}

	return fieldValuesAsStrings
}
```